src/Calypso-SystemPlugins-Reflectivity-Browser/ClyMetaLinkInstallationPresenter.class.st

```diff
@@ -0,0 +1,123 @@
+Class {
+	#name : #ClyMetaLinkInstallationPresenter,
+	#superclass : #SpPresenter,
+	#instVars : [
+		'sourceNode',
+		'list',
+		'toolBar',
+		'isForMetaLinkInstallation'
+	],
+	#category : #'Calypso-SystemPlugins-Reflectivity-Browser-Metalinks'
+}
+
+{ #category : #specs }
+ClyMetaLinkInstallationPresenter class >> defaultSpec [
+	^ SpBoxLayout newVertical
+		  add: #toolBar
+		  expand: false
+		  fill: false
+		  padding: 0;
+		  add: #list;
+		  yourself
+]
+
+{ #category : #'instance creation' }
+ClyMetaLinkInstallationPresenter class >> onNode: aSourceNode forInstallation: forInstallation [
+	^ self basicNew
+		  sourceNode: aSourceNode;
+		  isForMetaLinkInstallation: forInstallation;
+		  initialize
+]
+
+{ #category : #opening }
+ClyMetaLinkInstallationPresenter class >> openInstallerOnNode: aSourceNode [
+	(self onNode: aSourceNode forInstallation: true) openWithSpec
+]
+
+{ #category : #opening }
+ClyMetaLinkInstallationPresenter class >> openUninstallerOnNode: aSourceNode [
+	(self onNode: aSourceNode forInstallation: false) openWithSpec
+]
+
+{ #category : #accessing }
+ClyMetaLinkInstallationPresenter >> allRelevantMetaLinks [
+	| baseLinks |
+	baseLinks := self isForMetaLinkInstallation
+		             ifTrue: [ MetaLink allInstances ]
```

Comment:
This is not nice logic. It needs to be explicit registry of what can be installed instead of #allInstances call.

Reply:
Imagine somebody will run all metalink tests - there will be a lot of metalinks in memory (GC will not immediately remove them).

Reply:
This is temporary, I should have put a comment.
The idea would be to add an explicit option at the metalink level, so that only the ones that are "installable through the menu" would appear.
But we need to think about how to do that exactly.
I will add a comment, but this should not block: having that now is better than not having that.

Reply:
ok

Reply:
I added a comment.
I will also add an issueif this PR is merged.